### PR TITLE
Fix WalletconnectV2 changeAccount issues

### PR DESCRIPTION
**Pull request by walmat**
Fixes APP-1712

## What changed (plus any additional context for devs)
Fixes a few issues that were longstanding with walletconnect v2 connections. I noticed that certain dapps send over a `requiredNamespace:{}` so we were never entering the logically loops to handle switching accounts properly. Separate from that, and after I fixed that, the list item never updated the address shown until closing the `ConnectedDappsSheet` and reopening it. I added some react `useState` to fix this. 

Also, I noticed that the `chainIds` were referencing the `requiredNamespace` in `WalletConnectV2ListItem` so we weren't showing those for v2 connections.

## Screen recordings / screenshots

https://github.com/user-attachments/assets/2a4c6c1a-a5ea-4a9f-8338-97b664ac81ed

https://github.com/user-attachments/assets/2b015a9b-be28-43c7-b001-cbc54f482626

## What to test
- walletconnect v2 and v1 pipeline

### CODE REVIEW

File: src/walletConnect/index.tsx [hunk 1]
```diff
@@ -982,6 +981,7 @@ export async function addAccountToSession(session: SessionTypes.Struct, { addres
       const ns = session.namespaces[key];
 
       namespaces[key] = {
+        ...ns,
```
Comment (walmat) [inline src/walletConnect/index.tsx:984]: this keeps the required chainIds

File: src/walletConnect/index.tsx [hunk 2]
```diff
@@ -37,7 +37,7 @@ import { Box } from '@/design-system';
 import { AuthRequestAuthenticateSignature, AuthRequestResponseErrorReason, RPCMethod, RPCPayload } from '@/walletConnect/types';
 import { AuthRequest } from '@/walletConnect/sheets/AuthRequest';
 import { getProvider } from '@/handlers/web3';
-import { uniq } from 'lodash';
+import { isEmpty, uniq } from 'lodash';
```
Comment (walmat) [inline src/walletConnect/index.tsx:40]:
```suggestion
import { uniq } from 'lodash';
```

File: src/components/walletconnect-list/WalletConnectV2ListItem.tsx [hunk 3]
```diff
@@ -58,64 +60,49 @@ export function WalletConnectV2ListItem({ session, reload }: { session: SessionT
...
-    const approvalAccountInfo = getAccountProfileInfo(selectedWallet, walletNames, address);
-    return {
-      ...approvalAccountInfo,
-    };
-  }, [wallets, walletNames, address]);
-
   const handlePressChangeWallet = useCallback(() => {
     Navigation.handleAction(Routes.CHANGE_WALLET_SHEET, {
-      currentAccountAddress: address,
+      currentAccountAddress: accountInfo?.accountAddress,
```
Comment (walmat) [inline src/components/walletconnect-list/WalletConnectV2ListItem.tsx:94]:
```suggestion
      currentAccountAddress: address,
```

File: src/components/walletconnect-list/WalletConnectV2ListItem.tsx [hunk 4]
```diff
@@ -58,64 +60,49 @@ export function WalletConnectV2ListItem({ session, reload }: { session: SessionT
...
 
-      // defensive, just for types, should never happen
-      throw e;
+  useEffect(() => {
+    if (address) {
+      setAccountInfo(getAccountProfileInfo(findWalletWithAccount(wallets || {}, address), walletNames, address));
     }
+  }, [address, walletNames, wallets]);
+
+  const chains = useMemo(() => namespaces?.eip155?.chains || [], [namespaces]);
```
Comment (benisgold) [inline src/components/walletconnect-list/WalletConnectV2ListItem.tsx:78]: i dont think you need useMemo here

**Review by benisgold — APPROVED** [review]

**Comment by BrodyHughes** [comment]
Seems like when u switch wallets via rainbow it doesn’t disconnect the old one. unsure if this is expected behavior. To repro:
- go to https://react-app.walletconnect.com/
- connect with a wallet on eth, arb, op, and pol
- sign any txn on the site
- in rainbow go to connected apps
- switch the connected wallet
- go back to connected apps in rainbow
- we only show one wallet connected but the website shows 2 and you can sign txns with either wallet

**Review by BrodyHughes — APPROVED** [review]
my above concerns are expected behavior. tested against the dapp i linked:
- connecting multiple networks 
- doing eth sign
- sign typed data
- and send eth
- switched wallets in app and tested again.

File: src/components/walletconnect-list/WalletConnectV2ListItem.tsx (hunk 4 above)
Comment (walmat) [reply src/components/walletconnect-list/WalletConnectV2ListItem.tsx:78]: IDE was telling me I did. I'll leave it for now

**Pull request merged**